tests/security/apparmor/aa_genprof.pm

```diff
@@ -0,0 +1,110 @@
+# Copyright (C) 2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Test the profile generation utility of Apparmor
+# Maintainer: Wes <whdu@suse.com>
+# Tags: poo#36886, tc#1621140
+
+use strict;
+use base "consoletest";
+use testapi;
+use utils;
+use version_utils qw(is_sle is_leap is_tumbleweed);
+
+sub run {
+
+    my $aa_tmp_prof       = "/tmp/apparmor.d";
+    my $aa_genprof_allow  = qr/\(A\)llow.*\(D\)eny/m;
+    my $aa_genprof_scan   = qr/\(S\)can system/m;
+    my $aa_genprof_finish = $aa_genprof_scan;
+    my $aa_genprof_save   = qr/\(S\)ave Changes/m;
+    my $aa_genprof_filter = [$aa_genprof_allow, $aa_genprof_finish, $aa_genprof_save];
+
+    select_console 'root-console';
+
+    systemctl('restart apparmor');
+    systemctl('start auditd');
+
+    save_screenshot;
+
+    # Must disable stdout buffering to make pipe works
+    if (   is_sle('15+')
+        or is_leap('15.0+')
+        or is_tumbleweed)
+    {    # apparmor >= 2.8.95
+        assert_script_run "export PYTHONUNBUFFERED=1";
+    }
+    else {    # apparmor < 2.8.95
+        assert_script_run "sed -i '/use strict;/a \$|=1;' /usr/sbin/aa-genprof";
+    }
```

Comment:
Could you please turn around the if/else and avoid the explcit check for Tumbleweed as this should be our default anyway? E.g. something like

```perl
# apparmor < 2.8.95
if (is_sle('<15') || is_leap('<15.0')) {
…
}
else {
# default
…
}
```

Admittedly it's a matter of style so feel free to state that you do not want to change this unless you agree with my proposal

Reply:
Changed, thanks